src/SAML2/Assertion.php

```diff
@@ -1257,7 +1232,9 @@ public function getAttributes(): array
      */
     public function setAttributes(array $attributes): void
     {
-        $this->attributes = $attributes;
+        foreach ($attributes as $name => $value) {
+            $this->attributes[$name] = $value;
+        }
```

Comment:
The phpdoc here should change to reflect the actual value:

```php
     * @param Attribute[]
```

(and the same in `getAttributes()`, of course)